sdk/include/unwind.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#include <cdefs.h>
+#include <setjmp.h>
+
+/**
+ * On-stack linked list of cleanup handlers.
+ */
+struct CleanupList
+{
+	/// Next pointer.
+	CleanupList *next;
+	/// Jump buffer to return to.
+	__jmp_buf env;
+};
+
+/**
+ * Head of the cleanup list.
+ *
+ * This is stored in the space that the switcher reserves at the top of the
+ * stack.  The stack is zeroed on entry to a compartment and so this will be
+ * null until explicitly written to.
+ */
+__always_inline static inline struct CleanupList **cleanup_list_head()
+{
+	void     *csp = __builtin_cheri_stack_get();
```

Comment:
Does the first invocation of a thread entry go through the switcher path that reserves the space on stack? If not this might overwrite stack pointer and link register stored by the entry function. Maybe no big deal but perhaps unexpected.